sdk-extension/opentelemetry-sdk-extension-aws/tests/trace/propagation/test_aws_xray_format.py

```diff
@@ -0,0 +1,375 @@
+            AwsXRayPropagatorTest.carrier_getter, CaseInsensitiveDict()
+        )
+
+        self.assertEqual(
+            get_extracted_span_context(actual_context_encompassing_extracted),
+            INVALID_SPAN_CONTEXT,
+        )
+
+    def test_extract_not_sampled_context(self):
+        actual_context_encompassing_extracted = AwsXRayPropagatorTest.XRAY_PROPAGATOR.extract(
+            AwsXRayPropagatorTest.carrier_getter,
+            build_dict_with_xray_trace_header(),
+        )
+
+        self.assertEqual(
+            get_extracted_span_context(actual_context_encompassing_extracted),
+            get_extracted_span_context(build_test_context()),
+        )
+
+    def test_extract_sampled_context(self):
+        actual_context_encompassing_extracted = AwsXRayPropagatorTest.XRAY_PROPAGATOR.extract(
+            AwsXRayPropagatorTest.carrier_getter,
+            build_dict_with_xray_trace_header(sampled="1"),
+        )
+
+        self.assertEqual(
+            get_extracted_span_context(actual_context_encompassing_extracted),
+            get_extracted_span_context(
+                build_test_context(trace_flags=TraceFlags(TraceFlags.SAMPLED))
+            ),
+        )
+
+    def test_extract_different_order(self):
+        default_xray_trace_header_dict = build_dict_with_xray_trace_header()
+        trace_header_components = default_xray_trace_header_dict[
+            AwsXRayFormat.TRACE_HEADER_KEY
+        ].split(AwsXRayFormat.KV_PAIR_DELIMITER)
+        reversed_trace_header_components = AwsXRayFormat.KV_PAIR_DELIMITER.join(
+            trace_header_components[::-1]
+        )
+        xray_trace_header_dict_in_different_order = CaseInsensitiveDict(
+            {AwsXRayFormat.TRACE_HEADER_KEY: reversed_trace_header_components}
+        )
+        actual_context_encompassing_extracted = AwsXRayPropagatorTest.XRAY_PROPAGATOR.extract(
+            AwsXRayPropagatorTest.carrier_getter,
+            xray_trace_header_dict_in_different_order,
+        )
+
+        self.assertEqual(
+            get_extracted_span_context(actual_context_encompassing_extracted),
+            get_extracted_span_context(build_test_context()),
+        )
+
+    def test_extract_with_additional_fields(self):
+        default_xray_trace_header_dict = build_dict_with_xray_trace_header()
+        xray_trace_header_dict_with_extra_fields = CaseInsensitiveDict(
+            {
+                AwsXRayFormat.TRACE_HEADER_KEY: default_xray_trace_header_dict[
+                    AwsXRayFormat.TRACE_HEADER_KEY
+                ]
+                + ";Foo=Bar"
+            }
+        )
+        actual_context_encompassing_extracted = AwsXRayPropagatorTest.XRAY_PROPAGATOR.extract(
+            AwsXRayPropagatorTest.carrier_getter,
+            xray_trace_header_dict_with_extra_fields,
+        )
+
+        self.assertEqual(
+            get_extracted_span_context(actual_context_encompassing_extracted),
```

Comment:
And similarly, here it's hard to understand the test since it refers to this `actual_context_encompassing_extracted` and `build_test_context`, not sure what that is without a lot of jumping

Reply:
I changed it up a bit! I decided to keep `get_extracted_span_context` since it's a 1 line function which helps avoid repeated code.

I split `build_test_context` into `build_test_current_context` (to simplify the `inject` tests) and `build_test_span_context` because it is highly reused and hopefully easy to understand. I think it is cleaner than having the `SpanContext` constructor multiple times, but let me know what you think!